aea/test_tools/test_cases.py

```diff
@@ -481,7 +483,7 @@ def is_successfully_terminated(cls, *subprocesses: subprocess.Popen) -> bool:
         if not subprocesses:
             subprocesses = tuple(cls.subprocesses)
 
-        all_terminated = all([process.returncode == 0 for process in subprocesses])
+        all_terminated = all(process.returncode == 0 for process in subprocesses)
```

Comment:
why?

Reply:
minor optimization. Always use generator instead of list comprehension in `any` and `all` because otherwise you construct the array (time and memory waste) and lose the early termination advantage.